src/detect-base64-decode.c

```diff
@@ -87,19 +87,20 @@ int DetectBase64DecodeDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s
         payload_len -= data->offset;
     }
 
-    decode_len = MIN(payload_len, data->bytes);
-
+    uint32_t decode_len = MIN(payload_len, data->bytes);
 #if 0
     printf("Decoding:\n");
     PrintRawDataFp(stdout, payload, decode_len);
 #endif
 
-    uint32_t consumed = 0, num_decoded = 0;
-    (void)DecodeBase64(det_ctx->base64_decoded, det_ctx->base64_decoded_len_max, payload,
-            decode_len, &consumed, &num_decoded, Base64ModeRFC4648);
-    det_ctx->base64_decoded_len = num_decoded;
-    SCLogDebug("Decoded %d bytes from base64 data.",
-        det_ctx->base64_decoded_len);
+    if (decode_len > 0) {
+        uint32_t num_decoded = 0;
+        (void)Base64Decode(payload, decode_len, det_ctx->base64_decoded_len_max, Base64ModeRFC4648,
+                det_ctx->base64_decoded, &num_decoded);
+        DEBUG_VALIDATE_BUG_ON(num_decoded >= decode_len);
```

Comment:
Why this debug validation ?

Reply:
By the way, is there a plan to deprecate this keyword ? (in the favor of transform)

Reply:
> By the way, is there a plan to deprecate this keyword ? (in the favor of transform)

No idea.. @jlucovsky ?

Reply:
Should then be done in `Base64Decode` itself, right ?

Reply:
indeed. Would be the better place.